modules/casper.js

```diff
@@ -214,6 +216,55 @@ var Casper = function Casper(options) {
         */
     });
 
+    // management ResourceTimeout with slimerJS
+    if (phantom.casperEngine === "slimerjs") {
+        this.checkTimeout = (function (data) {
+            var self = data;
+            return function(){
+                if ((!!this.page.settings.resourceTimeout) && (!isNaN(this.page.settings.resourceTimeout))) {
+                    var d = new Date().getTime();
+                    for (var i = self.resourcesTime.length-1 ; i >= 0 ; i--) {
+                        if (d > self.resourcesTime[i].time + self.page.settings.resourceTimeout) {
+                            var request = self.resourcesTime[i];
+                            self.resourcesTime.splice(i, 1);
+                            request.errorCode = 408;
+                            request.errorString = "Network timeout on resource.";
+                            request.abort();
+                            self.page.onResourceTimeout(request);
+                        }
+                    }
+                }
+            };
+        })(this);
+        this.on('resource.requested', function(requestData, networkRequest) {
+            var obj = { 
+                "id": requestData.id,
+                "time": new Date(requestData.time).getTime(),
+                "abort": (function(network){ var ab = network; return function(){ ab.abort();};
+                })(networkRequest)
+            };
+            this.resourcesTime.push(obj);
+        });
+        this.on('resource.received', function(resource) {
+                if (resource.stage === "end") {
+                    var nbResource = this.resourcesTime.length;
+                    this.resourcesTime = this.resourcesTime.filter(function(elm){
+                        return elm.id !== resource.id;
+                    });
+                    if (resource.body === "" && nbResource !== this.resourcesTime.length){
+                        resource.errorCode = 408;
+                        resource.errorString = "Network timeout on resource.";
+                        this.page.onResourceTimeout(resource);
+                    }
+                }
+        });
+        this.on('resource.error', function(resourceError) {
+            this.resourcesTime = this.resourcesTime.filter(function(resource){
+                return resource.id !== resourceError.id;
+            });
+        });
+    }
```

Comment:
Does it means `resource.timeout` event is not yet available for PhantomJS ? My question sounds stupid, but I want to be sure before consider merging :)